integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/hotdeployment/test/SynapseArtifactsHotDeploymentTestCase.java

```diff
@@ -0,0 +1,164 @@
+
+    private final String proxyName = "HotDeploymentTestProxy";
+    private final String proxyFileName = "HotDeploymentTestProxy.xml";
+
+    private final String sequenceName = "HotDeploymentTestSequence";
+    private final String sequenceFileName = "HotDeploymentTestSequence.xml";
+
+    private final String endpointName = "HotDeploymentTestEndpoint";
+    private final String endpointFileName = "HotDeploymentTestEndpoint.xml";
+
+    private LogViewerClient logViewerClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void deployService() throws Exception {
+        super.init();
+        copyArtifactsToDeploymentDirectory();
+        logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+    }
+
+    @Test(groups = "wso2.esb",
+          description = "Proxy Service Hot Deployment")
+    public void testProxyServiceHotDeployment() throws Exception {
+        String proxyServiceFile = SERVER_DEPLOYMENT_DIR + "proxy-services" + File.separator + proxyFileName;
+
+        Assert.assertTrue(esbUtils.isProxyDeployed(contextUrls.getBackEndUrl(), sessionCookie, proxyName),
+                "Proxy Deployment failed");
+
+        logViewerClient.clearLogs();
+        FileUtils.touch(new File(proxyServiceFile));
+        log.info(proxyFileName + " has been updated and waiting for redeployment");
+        Assert.assertTrue(searchInLogs(logViewerClient, "'HotDeploymentTestProxy' has been update from file"),
+                "Proxy deployment failed on updating file");
+        FileManager.deleteFile(proxyServiceFile);
+        Assert.assertTrue(esbUtils.isProxyUnDeployed(contextUrls.getBackEndUrl(), sessionCookie, proxyName),
+                "Proxy Undeployment failed");
+    }
+
+    @Test(groups = "wso2.esb",
+          description = "Sequence Hot Deployment")
+    public void testSequenceHotDeployment() throws Exception {
+        String sequenceFile = SERVER_DEPLOYMENT_DIR + "sequences" + File.separator + sequenceFileName;
+
+        Assert.assertTrue(esbUtils.isSequenceDeployed(contextUrls.getBackEndUrl(), sessionCookie, sequenceName),
+                "Sequence Deployment failed");
+
+        logViewerClient.clearLogs();
+        FileUtils.touch(new File(sequenceFile));
+        log.info(sequenceFileName + " has been updated and waiting for redeployment");
+        Assert.assertTrue(searchInLogs(logViewerClient, "HotDeploymentTestSequence has been updated from the file"),
+                "Sequence deployment failed on updating file");
+
+        FileManager.deleteFile(sequenceFile);
+        Assert.assertTrue(esbUtils.isSequenceUnDeployed(contextUrls.getBackEndUrl(), sessionCookie, proxyName),
+                "Sequence Undeployment failed");
+    }
+
+    @Test(groups = "wso2.esb",
+          description = "Endpoint Hot Deployment")
+    public void testEndpointHotDeployment() throws Exception {
+        String endpointFile = SERVER_DEPLOYMENT_DIR + "endpoints" + File.separator + endpointFileName;
```

Comment:
Rather than depending on the logs, can't we simply send a request to the service and check whether deployment has happend ?

Reply:
Here we are checking the redeployment, that is why log is asserted